### PR TITLE
Update FoodChainTestGenerator. Update FoodChainTest. Refs #370, #331

### PR DESCRIPTION
**Pull request by ricemery**
* Update FoodChainTestGenerator. 
* Update FoodChainTest.
* Update example.scala to support returning either a single verse or a range of verses.
 
Refs #370, #331

### CODE REVIEW

File: exercises/food-chain/example.scala [hunk 1]
```diff
@@ -29,5 +29,7 @@ object FoodChain {
       })
   }
 
-  lazy val song: String = (for {i <- List.range(0, parts.length)} yield {verse(i)}).mkString
+  def verse(n: Int): String = verseInternal(n - 1)
+
+  def verse(n: Int, m: Int): String = (for {i <- List.range(n - 1, m)} yield {verseInternal(i)}).mkString
```
Comment (ErikSchierboom) [inline exercises/food-chain/example.scala]: I would expect

**Comment by ricemery** [comment]
Hello Erik:

I'm not sure what you were commenting on in example.scala ->  "I would
expect"?

Thanks

**Comment by ErikSchierboom** [comment]
That was a mistake I made. I first wanted to comment on the necessity of a helper function, but then decided not to. Apparently it got published, so I deleted it afterwards. Sorry about that

**Comment by ricemery** [comment]
Cool. Thanks.

**Pull request merged**